From 3f8f6c077c6b79dedb316732867867f2e135174b Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Fri, 31 Jul 2015 16:03:25 +0200
Subject: [PATCH 2/4] safe renegotiation: handle case where client didn't send
 any extension

That was affected by the "don't try to send extensions we didn't receive".
---
 lib/ext/safe_renegotiation.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/ext/safe_renegotiation.c b/lib/ext/safe_renegotiation.c
index 314c4e2..1a58418 100644
--- a/lib/ext/safe_renegotiation.c
+++ b/lib/ext/safe_renegotiation.c
@@ -60,6 +60,10 @@ _gnutls_ext_sr_finished(gnutls_session_t session, void *vdata,
 					   &epriv);
 	if (ret < 0) {
 		gnutls_assert();
+		/* if a client didn't advertise safe renegotiation, we treat
+		 * it as disabled. */
+		if (session->security_parameters.entity == GNUTLS_SERVER)
+			return 0;
 		return ret;
 	}
 	priv = epriv.ptr;
-- 
2.4.6

